models/profiles-ml.yaml

```diff
@@ -1,140 +1,139 @@
 models:
-  - name: &model_name_7_days churn_7_days_model
-    model_type: python_model
+  - name: churn_7_days_model
+    model_type: propensity
     model_spec:
-      occurred_at_col: insert_ts
       entity_key: user
-      py_repo_url: https://github.com/rudderlabs/rudderstack-profiles-classifier.git
-      train:
-        file_extension: .json
-        file_validity: 168h # If the last trained model is older than this, then the model will be trained again,
-        inputs: &inputs_7_days
-          - models/rudder_user_base_features # inputs to materialise the required data for model
-        config:
-          data: &model_data_configs_7_days
-            label_column: is_churned_7_days # Name of column for which we want to find predictions
-            label_value: 1 # Value of label column for which prediction is generated
-            prediction_horizon_days: 7 # Number of days in future for which we want to predict
-            output_profiles_ml_model: *model_name_7_days # Name of output model based on current model to dinstinguish between multiple models
-            eligible_users:
-            inputs:
-              - packages/base_features/models/rudder_user_base_features
-          preprocessing: &model_prep_configs_7_days
-            ignore_features:
-              - user_email
-              - first_name
-              - last_name
-      predict:
-        inputs:
-          - models/rudder_user_base_features
-        config:
-          data: *model_data_configs_7_days # Use the same config as training
-          preprocessing: *model_prep_configs_7_days # Use the same config as training
-          outputs:
-            column_names:
-              percentile: &percentile_name_7_days percentile_churn_score_7_days # Name of column in output table having percentile score
-              score: churn_score_7_days # Name of column in output table having probabilistic score
-            feature_meta_data: &feature_meta_data_7_days
-              features:
-                - name: *percentile_name_7_days
-                  description: "Percentile of churn score. Higher the percentile, higher the probability of churn"
-      <<: *feature_meta_data_7_days
-      contract:
-        is_optional: false
-        is_event_stream: true
-        with_entity_ids: []
-        with_columns:
-          - name: percentile_churn_score_7_days
-  - name: &model_name_30_days churn_30_days_model
-    model_type: python_model
+      training:
+        predict_var: entity/user/is_churned_7_days
+        predict_window_days: 7
+        label_value: 1
+      prediction:
+        output_columns:
+          percentile:
+            name: percentile_churn_score_7_days
+            description: "Percentile of churn score. Higher the percentile, higher the probability of churn"
+          score:
+            name: churn_score_7_days
+            is_feature: False
+      inputs:
+        - entity/user/days_since_last_seen
```

Comment:
@dpatchigolla Do we want to cut down the number of inputs for the model ? Unlike the feature table model as in input, if an entity var is disabled, it will disable the propensity model.

Reply:
Yes. Let's bring this down to following:
```
        - days_since_last_seen
        - days_since_account_creation
        - active_days_in_past_7_days
        - active_days_in_past_365_days
        - total_sessions_till_date
        - total_sessions_last_week
        - total_sessions_90_days
        - total_sessions_365_days
        - avg_session_length_in_sec_overall
        - avg_session_length_in_sec_last_week
        - avg_session_length_in_sec_365_days
        - is_churned_7_days
        - is_churned_30_days
        - is_churned_90_days
```

This skips the categorical and array type features, which are typically more noisy. The above features all rely only on pages/tracks and shouldn't usually cause errors either.

Reply:
Updated